src/dbnode/encoding/proto/custom_marshal.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package proto
+
+import (
+	"math"
+
+	"github.com/golang/protobuf/proto"
+)
+
+type customFieldMarshaler interface {
+	encFloat64(tag int32, x float64)
+	encFloat32(tag int32, x float32)
+	encInt32(tag int32, x int32)
+	encSInt32(tag int32, x int32)
```

Comment:
I assume the S variants are `varint` or `short`?  Maybe use `encVarInt32` or `encShortInt32` instead?  

The single letter S is strange and not as clear the delineation while reading the API.

Reply:
Yeah I can see how that would be confusing. It actually stands for "signed" and these names are basically verbatim what they're called in in the Protobuf specification.

I'll add a comment